### PR TITLE
Only take inter projects into account in compile config

### PR DESCRIPTION
**Pull request by alexarchambault**
Fixes https://github.com/coursier/coursier/issues/1340.

### CODE REVIEW

File: modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala [hunk 1]
```diff
@@ -76,14 +83,14 @@ object ResolutionRun {
...
           }
         )
-        .withRepositories(params.repositories)
+        .withRepositories(repositories)
         .withResolutionParams(
           params
             .params
-            .addForceVersion(params.interProjectDependencies.map(_.moduleVersion): _*)
-            .withForceScalaVersion(isCompileConfig && params.autoScalaLibOpt.nonEmpty)
+            .addForceVersion((if (isCompileConfig) params.interProjectDependencies.map(_.moduleVersion) else Nil): _*)
```
Comment (eed3si9n) [inline modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala]: Doesn't it matter what the scoping of the `interProjectDependencies` are? You could have `dependsOn` scoped to `Test` for example.

Comment (alexarchambault) [reply modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala]: Test should be fine, that actually includes all the configs in the config sub-graph of compile (that is, runtime / test are included at least, scala-tool isn’t connected to compile, so isn’t included).

Comment (alexarchambault) [reply modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala]: We run one resolution per config sub-graph.

Comment (eed3si9n) [reply modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala]: Someone could also scope to some other sandbox configuration like `.dependsOn(something % "bar->bar")`

Comment (alexarchambault) [reply modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala]: Would you feel like submitting a fix for that? (checking which configurations are used in inter-project deps, and passing that around and using it around the code touched by this PR?

Comment (eed3si9n) [reply modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala]: I might come around to it eventually, but probably I won't be able to soon.

**Pull request merged**